src/rest.cpp

```diff
@@ -590,45 +590,31 @@ static bool rest_chaininfo(const std::any& context, HTTPRequest* req, const std:
     }
 }
 
-static bool rest_mempool_info(const std::any& context, HTTPRequest* req, const std::string& strURIPart)
+static bool rest_mempool(const std::any& context, HTTPRequest* req, const std::string& str_uri_part)
 {
     if (!CheckWarmup(req))
         return false;
-    const CTxMemPool* mempool = GetMemPool(context, req);
-    if (!mempool) return false;
-    std::string param;
-    const RESTResponseFormat rf = ParseDataFormat(param, strURIPart);
-
-    switch (rf) {
-    case RESTResponseFormat::JSON: {
-        UniValue mempoolInfoObject = MempoolInfoToJSON(*mempool);
 
-        std::string strJSON = mempoolInfoObject.write() + "\n";
-        req->WriteHeader("Content-Type", "application/json");
-        req->WriteReply(HTTP_OK, strJSON);
-        return true;
-    }
-    default: {
-        return RESTERR(req, HTTP_NOT_FOUND, "output format not found (available: json)");
-    }
+    std::string param;
+    const RESTResponseFormat rf = ParseDataFormat(param, str_uri_part);
+    if (param != "contents" && param != "info") {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Invalid URI format. Expected /rest/mempool/<info|contents>.json");
     }
-}
 
-static bool rest_mempool_contents(const std::any& context, HTTPRequest* req, const std::string& strURIPart)
-{
-    if (!CheckWarmup(req)) return false;
     const CTxMemPool* mempool = GetMemPool(context, req);
     if (!mempool) return false;
-    std::string param;
-    const RESTResponseFormat rf = ParseDataFormat(param, strURIPart);
 
     switch (rf) {
     case RESTResponseFormat::JSON: {
-        UniValue mempoolObject = MempoolToJSON(*mempool, true);
+        std::string str_json;
+        if (param == "contents") {
+            str_json = MempoolToJSON(*mempool, true).write() + "\n";
+        } else {
+            str_json = MempoolInfoToJSON(*mempool).write() + "\n";
+        }
```

Comment:
Maybe reduce temporaries and reassignments here.

```diff
     case RESTResponseFormat::JSON: {
-        std::string str_json;
-        if (param == "contents") {
-            str_json = MempoolToJSON(*mempool, true).write() + "\n";
-        } else {
-            str_json = MempoolInfoToJSON(*mempool).write() + "\n";
-        }
-
         req->WriteHeader("Content-Type", "application/json");
-        req->WriteReply(HTTP_OK, str_json);
+        req->WriteReply(HTTP_OK, (param == "contents" ? MempoolToJSON(*mempool, /*verbose=*/true) : MempoolInfoToJSON(*mempool)).write() + "\n");
         return true;
```

Reply:
It could make sense to avoid `std::string` copies in `WriteReply`, but maybe for a follow-up?